analyzers/tests/SonarAnalyzer.UnitTest/TestCases/NameOfShouldBeUsed.CSharp11.cs

```diff
@@ -0,0 +1,40 @@
+﻿using System;
+
+namespace Tests.Diagnostics
+{
+    class Program
+    {
+        public void Method_With_RawStringLiterals(int arg1)
+        {
+            // Repro for https://github.com/SonarSource/sonar-dotnet/issues/6441
+            if (arg1 < 0)
+                throw new Exception("""arg1"""); // FN
+            else if (arg1 > 100)
+                throw new ArgumentException("""Bad parameter name""", """arg1"""); // FN
+        }
+
+
+        public void Method_With_NewLinesInStringInterpolation(int arg1)
+        {
+            string argName = "arg1";
+            if (arg1 < 0)
+            {
+                throw new Exception($"{
+                    arg1 switch
+                    {
+                        < 0 => "arg1",  // Noncompliant
+                        _ => "Can't touch this",
+                    }}");
+            }
+            if (arg1 > 0)
+            {
+                const string arg1Name = """arg1""";
+                throw new ArgumentException("Message" ,$"{arg1Name}"); // FN
```

Comment:
@martin-strecker-sonarsource  Why this is not an FN?

Reply:
Because you could do this:
```cs
const string paramName = nameof(param);
if (param<0) throw new ArgumentException("Must be greater than 0", paramName);
if (param>100) throw new ArgumentException("Must be smaller than 100", paramName);
```
You need to do some kind of flow analysis to detect that paramName was set using `nameof`. In line two and three you can ask the compiler for the constant value of `paramName` but you would just get "param" but not that is was set using `nameof`. It is possible to track this down, but not in the scope of the rule.

Reply:
Cool! Thanks a lot for the explanation.